### PR TITLE
test: stop using npm link in CLI tests

### LINKED ISSUE #245: Test failures
While working a little bit on linkinator on my Windows dev VM I notice an issue if linkinator is already installed globally:

```
  1) cli
       "before all" hook for "should show output for failures":
     Error: Command failed with exit code 1: npm link
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code EEXIST
npm ERR! path C:\Users\xmr\AppData\Roaming\npm\node_modules\linkinator\build\src\cli.js
npm ERR! dest C:\Users\xmr\AppData\Roaming\npm\linkinator.cmd
npm ERR! EEXIST: file already exists, cmd shim 'C:\Users\xmr\AppData\Roaming\npm\node_modules\linkinator\build\src\cli.js' -> 'C:\Users\xmr\AppData\Roaming\npm\linkinator.cmd'
npm ERR! File exists: C:\Users\xmr\AppData\Roaming\npm\linkinator.cmd
npm ERR! Remove the existing file and try again, or run npm
npm ERR! with --force to overwrite files recklessly.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\xmr\AppData\Roaming\npm-cache\_logs\2021-01-19T07_04_17_258Z-debug.log

> linkinator@0.0.0 prepare C:\Users\xmr\Desktop\linkinator
> npm run compile


> linkinator@0.0.0 compile C:\Users\xmr\Desktop\linkinator
> tsc -p .

audited 1125 packages in 12.906s

80 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

      at makeError (node_modules\execa\lib\error.js:59:11)
      at handlePromise (node_modules\execa\index.js:114:26)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at async Context.<anonymous> (build\test\test.cli.js:18:9)
          -> C:\Users\xmr\Desktop\linkinator\test\test.cli.ts:19:5
```

```
C:\Users\xmr\Desktop\linkinator>node -v && npm -v
v14.15.4
6.14.10
```

It seems after running `npm t`, linkinator is being installed globally, so subsequent runs fail.

**EDIT**: the root cause must be this https://github.com/JustinBeckwith/linkinator/blob/071a220712aded6325cab827b2ac9a4deb1a4940/test/test.cli.ts#L18-L20

I'd say it'd be best if this was removed completely. That would also speed tests up. Otherwise, there should be an `unlink` after tests are done.

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Fixes #245 maybe.  Makes @XhmikosR and their windows machine happier 🙃 Only risk here is not entirely covering the way CLI is tested with the package.

### CODE REVIEW

File: test/test.cli.ts [hunk 1]
```diff
@@ -6,18 +6,8 @@ import * as util from 'util';
 import enableDestroy = require('server-destroy');
 import {LinkResult, LinkState} from '../src/index';
 
-// eslint-disable-next-line prefer-arrow-callback
-describe('cli', function () {
+describe('cli', () => {
   let server: http.Server;
-  this.timeout(60_000);
```
Comment (XhmikosR) [inline test/test.cli.ts:12]: I think the default timeout is not enough for the Windows CI :)

**Comment by XhmikosR** [comment]
@JustinBeckwith thanks for looking into this!

With this branch I'm getting the same failures as on CI locally.

With the following patch it works for me: https://github.com/XhmikosR/linkinator/commit/fb8c97279beec16e50db06dfdf15d056851a75ef

<details><summary>npm t</summary>

```
C:\Users\xmr\Desktop\linkinator>npm t

> linkinator@0.0.0 pretest C:\Users\xmr\Desktop\linkinator
> npm run compile


> linkinator@0.0.0 compile C:\Users\xmr\Desktop\linkinator
> tsc -p .


> linkinator@0.0.0 test C:\Users\xmr\Desktop\linkinator
> c8 mocha build/test



  cli
    √ should show output for failures (5151ms)
    √ should pass successful markdown scan (576ms)
    √ should allow multiple paths (594ms)
    √ should show help if no params are provided (489ms)
    √ should flag skipped links (608ms)
    √ should provide CSV if asked nicely (594ms)
    √ should provide JSON if asked nicely (584ms)
    √ should not show links if --silent (617ms)
    √ should not show 200 links if verbosity is ERROR with JSON (634ms)
    √ should accept a server-root (588ms)
    √ should accept globs (641ms)
    √ should throw on invalid format (493ms)
    √ should throw on invalid verbosity (500ms)
    √ should throw when verbosity and silent are flagged (490ms)
    √ should show no output for verbosity=NONE (5163ms)
    √ should show callstacks for verbosity=DEBUG (5085ms)
    √ should allow passing a config (556ms)
    √ should warn on retries (1613ms)

  config
    √ should allow passing no config
Unable to find config file /path/does/not/exist
    √ should throw with a reasonable message if the path doesnt exist
    √ should allow reading from a config file
    √ should merge config settings from the CLI and object

  linkinator
    √ should perform a basic shallow scan (45ms)
    √ should only try a link once (57ms)
    √ should only queue a link once (39ms)
    √ should skip links if asked nicely
    √ should skip links if passed a linksToSkip function
    √ should report broken links
    √ should handle relative links (52ms)
    √ should handle fetch exceptions
    √ should report malformed links as broken (54ms)
    √ should detect relative urls with relative base (404ms)
    √ should detect relative urls with absolute base (94ms)
    √ should detect broken image links
    √ should perform a recursive scan (49ms)
    √ should not recurse non-html files
    √ should not follow non-http[s] links
    √ should not recurse by default
    √ should retry with a GET after a HEAD (92ms)
    √ should only follow links on the same origin domain (89ms)
    √ should not attempt to validate preconnect or prefetch urls (61ms)
    √ should attempt a GET request if a HEAD request fails on external links (130ms)
    √ should support a configurable timeout
    √ should handle markdown (52ms)
    √ should throw an error if you pass server-root and an http based path
    √ should allow overriding the server root (40ms)
    √ should accept multiple filesystem paths
    √ should not allow mixed local and remote paths
    √ should require at least one path
    √ should not pollute the original options after merge
    √ should accept multiple http paths (119ms)
    √ should print debug information when the env var is set (41ms)
    √ should respect globs (57ms)
    √ should autoscan markdown if specifically in path
    √ should throw if a glob provides no paths to scan
    √ should always send a human looking User-Agent (127ms)
    √ should surface call stacks on failures in the API
    √ should respect server root with globs (69ms)
    √ should respect absolute server root (60ms)
    √ should scan links in <meta content="URL"> tags (82ms)
    √ should support directory index
    √ should disabling directory index by default (47ms)
    √ should provide a relative path in the results (51ms)
    √ should provide a server root relative path in the results (53ms)

  retries
    √ should handle 429s with invalid retry-after headers (51ms)
    √ should retry 429s with second based header (136ms)
    √ should retry 429s after failed HEAD (121ms)
    √ should retry 429s with date based header (126ms)
    √ should detect requests to wait on the same host (119ms)
    √ should increase timeout for followup requests to a host (95ms)


  70 passing (28s)

------------|---------|----------|---------|---------|-------------------
File        | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
------------|---------|----------|---------|---------|-------------------
All files   |   99.21 |    96.71 |     100 |   99.21 |
 cli.ts     |     100 |      100 |     100 |     100 |
 config.ts  |   96.08 |    90.91 |     100 |   96.08 | 43-44
 index.ts   |   99.57 |    95.87 |     100 |   99.57 | 327-328
 links.ts   |    96.8 |    91.67 |     100 |    96.8 | 53-54,98-99
 logger.ts  |     100 |      100 |     100 |     100 |
 options.ts |     100 |    97.62 |     100 |     100 | 134
 queue.ts   |   97.33 |       95 |     100 |   97.33 | 52-53
 server.ts  |     100 |      100 |     100 |     100 |
------------|---------|----------|---------|---------|-------------------

```

</details>

I'm unsure if 10 seconds will be enough for CI too, but it's worth trying that.

Great time decrease, the running system is not changed during tests, and no random errors on Windows :)

File: test/test.cli.ts [hunk 2]
```diff
@@ -26,34 +16,43 @@ describe('cli', function () {
   });
 
   it('should show output for failures', async () => {
-    const res = await execa('linkinator', ['test/fixtures/basic'], {
-      reject: false,
-    });
+    const res = await execa(
+      'node',
+      ['build/src/cli.js', 'test/fixtures/basic'],
```
Comment (XhmikosR) [inline test/test.cli.ts:21]: Have you thought about using package.json's `bin` property?

Comment (JustinBeckwith) [reply test/test.cli.ts:21]: Huh - suppose I could.  Would the idea be to just make sure we have the right value in package.json?

Comment (XhmikosR) [reply test/test.cli.ts:21]: Yup. I thought that's what you meant by "Only risk here is not entirely covering the way CLI is tested with the package." but maybe you meant something else.

**Comment by XhmikosR** [comment]
There's only one `link` instance AFAICT in the npm `docs-test` script but since it's not part of `npm test` it doesn't seem to cause any issues.

File: test/test.cli.ts (hunk 1 above)
Comment (JustinBeckwith) [reply test/test.cli.ts:12]: Goodness gravy.

File: test/test.cli.ts (hunk 2 above)
Comment (JustinBeckwith) [reply test/test.cli.ts:21]: Nah you're right, that reduces some of the risk.  Good call!

**Pull request merged**